discovery/eureka_test.go

```diff
@@ -0,0 +1,119 @@
+package discovery
+
+import (
+	"gopkg.in/ory-am/dockertest.v3"
+	dc "github.com/fsouza/go-dockerclient"
+	"github.com/jaume-pinyol/fargo"
+	"log"
+	"os"
+	"testing"
+	"strconv"
+	"github.com/op/go-logging"
+)
+
+var eurekaTestPort int = 8080
+var eurekaTestUrl string = "http://127.0.0.1:" + strconv.Itoa(eurekaTestPort) + "/eureka"
+
+func TestMain(m *testing.M) {
+	// uses a sensible default on windows (tcp/http) and linux/osx (socket)
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		log.Fatalf("Could not connect to docker: %s", err)
+	}
+	// pulls an image, creates a container based on it and runs it
+	resource, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository:   "netflixoss/eureka",
+		Tag:          "1.3.1",
+		PortBindings: map[dc.Port][]dc.PortBinding{
+			dc.Port(strconv.Itoa(eurekaTestPort) + "/tcp"): {{HostIP: "", HostPort: strconv.Itoa(eurekaTestPort)}},
+		},
+	})
+	if err != nil {
+		log.Fatalf("Could not start resource: %s", err)
+	}
+
+	// exponential backoff-retry, because the application in the container might not be ready to accept connections yet
+	if err := pool.Retry(func() error {
+		_, err := NewEurekaClient(eurekaTestUrl)
+		return err
+	}); err != nil {
+		log.Fatalf("Could not connect to the docker resource: %s", err)
+	}
+
+	code := m.Run()
```

Comment:
what is Run from Testing doing?

Reply:
See the previous reference about what TestMain actually does